src/node.cc

```diff
@@ -4554,19 +4557,14 @@ inline int Start(Isolate* isolate, IsolateData* isolate_data,
     SealHandleScope seal(isolate);
     bool more;
     do {
-      v8_platform.PumpMessageLoop(isolate);
-      more = uv_run(env.event_loop(), UV_RUN_ONCE);
-
-      if (more == false) {
-        v8_platform.PumpMessageLoop(isolate);
-        EmitBeforeExit(&env);
-
-        // Emit `beforeExit` if the loop became alive either after emitting
-        // event, or after running some callbacks.
-        more = uv_loop_alive(env.event_loop());
-        if (uv_run(env.event_loop(), UV_RUN_NOWAIT) != 0)
-          more = true;
-      }
+      uv_run(env.event_loop(), UV_RUN_DEFAULT);
```

Comment:
Is everyone here sure this doesn't break something edge-casey with beforeExit and order of operations?

I'm trying to dig into it but have pretty much run out of time this week.

It seems to me that in some case(s) before, uv callbacks may have been called twice before the next `beforeExit`, which could alter things at the very least like an addon that uses a `uv_check_t` handle?